README.md

```diff
@@ -1 +1,110 @@
+
+## Objectives
+
+- Define MVC (Model-View-Controller), MVVM (Model-View-ViewModel) as they relate to both Single Page Applications and software programming in general
+- Describe one-way and two-way data-binding that Angular utilizes and offers
+- Describe Dependency Injection, a clever way to enable us to have what we want, wherever we want it
+- Describe a Single Page Application and common features
+- Demonstrate server-side rendering concepts
+
+## Single Page Applications
+
+Angular allows us to create single page applications (SPAs). The means that the user goes to one page and very rarely will have to refresh/leave that page to go to other parts of the website. Ever wondered how we can go to Twitter and never have to actually refresh the page? Yet all of our notifications, tweets, etc are all real-time and constantly up-to-date? You guessed it - Twitter is a single page application!
+
+Single page applications handle all the routing, calling different parts of your codebase dependant on what URL the user is currently at. If we look back at Twitter, you'll notice how the URL does change page yet the page doesn't refresh. This is because we have now handled all routing over to the client side instead of the server. The client can see where the user has intended to go, and select the specific code for that page - magical!
+
+Tranditionally, server-side rendering would be like the following -
+
+![Server-side rendering](http://www.michaelgallego.fr/images/posts/2012-11-26-client-side-1.png)
+
+The user requests the page from the server, and the server will return all the HTML to display on that page. The data would not be updated until the user refreshes. This puts overhead on the server, as it will have to render the page on every page load. If you head over to the [BBC News](http://bbc.co.uk/news) you'll see how the content is already there - no waiting whatsoever.
+
+With Angular, the flow will go like this instead -
+
+![Client-side rendering](http://branchandbound.net/pics/client-side-rendering-only.png)
+
+This looks a lot more complicated, but it isn't! Instead of returning rendered HTML for each user, the server would return the same page for every user. The client will then make a request to the server, requesting what's necessary for that page to render (such as a list of products). The client will then render this page for the user.
+
+This has massive advantages - if 1000 users are on the website, instead of one server rendering 1000 pages, 1000 clients are rendering one page, reducing strain on the server. We can also re-use backend endpoints - our mobile website and our desktop website can both request the same data from the server but display it differently.
+
+One example of this that you can checkout for yourselves is if you go to Facebook. You might notice how you don't actually see anything until the content very quickly and then the client renders this - saving so many resources in the mean time!
```

Comment:
Great example, but the wording here is a little bit strange. Can you rephrase this?